### PR TITLE
Catch for NoClasDefFoundError

### PR DESCRIPTION
**Pull request by makiam**
I try to install GLRenderer plugin with SPManager. 
Once application is restarted it stuck on showing TitleWindow.
I see NoClassDefFoundError as this plugin expected classes from old JOGL

### CODE REVIEW

**Comment by Sailsman63** [comment]
Not sure why this error is being thrown, rather than `ClassNotfoundException`

Perhaps our load routine needs a little work. I think this is probably a little to targeted as a fix. Next time you find something like this, please file it as a bug report.

**Comment by Sailsman63** [comment]
@peastman how would you feel about catching all (`LinkageError`s)[https://docs.oracle.com/javase/7/docs/api/java/lang/LinkageError.html] here, just to insulate ourselves from any sort of oops that the plugin code generates?

I'd also like to see the Plugin repo updated to mark the current version of `GLRenderer` as incompatible with 3.0.3 and up, as there is a certain type of user that always starts every new install with 'doonload all teh pluginz'

**Comment by peastman** [comment]
That sounds reasonable.  I wonder if we should just catch `Error`?  Even if it's something really strange that leaves the JVM in a broken state, we might at least be able to display a message telling the user which plugin is causing the problem.

File: ArtOfIllusion/src/artofillusion/PluginRegistry.java [hunk 1]
```diff
@@ -203,6 +204,12 @@ private static void processJar(JarInfo jar, Map<String, JarInfo> nameMap)
         registerResource(info.type, info.id, jar.loader, info.name, info.locale);
       }
     }
+    catch(NoClassDefFoundError ncdfe)
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/PluginRegistry.java:207]: As previously mentioned, I think we do need to broaden this, at least to `LinkageError` if not all the way to just `Error`.

File: ArtOfIllusion/src/artofillusion/PluginRegistry.java [hunk 2]
```diff
@@ -203,6 +204,12 @@ private static void processJar(JarInfo jar, Map<String, JarInfo> nameMap)
...
+    {
+      new BStandardDialog("", UIUtilities.breakString(Translate.text("pluginLoadError", jar.file.getName())), BStandardDialog.ERROR).showMessageDialog(null);
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/PluginRegistry.java:209]: As previously mentioned, I think we do need to broaden this, at least to `LinkageError` if not all the way to just `Error`.

**Review by Sailsman63 — CHANGES_REQUESTED** [review]

File: ArtOfIllusion/src/artofillusion/PluginRegistry.java (hunk 2 above)
Comment (Sailsman63) [reply ArtOfIllusion/src/artofillusion/PluginRegistry.java:209]: Thanks for the update. I did some further testing with an (easy to make) broken plugin. We do need to just broaden all the way to `Error` here.

I can make the change myself, if you prefer.

**Review by Sailsman63 — APPROVED** [review]

**Comment by Sailsman63** [comment]
Thank you!

**Pull request merged**